### PR TITLE
[DEV-1925]: Causely Slack Bot

### PR DESCRIPTION
**Pull request by sumanyumuku98**
# Description

Helm chart to install the causely bot to forward the notification payloads to external endpoints like slack, discord, pagerduty e.t.c.

JIRA: [DEV-1925]

[DEV-1925]: https://causely.atlassian.net/browse/DEV-1925?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: helm/causelybot/templates/bot_deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{{- include "validateImageRepository" . -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: causelybot
+  namespace: {{ .Release.Namespace }}
```
Comment (esara) [inline helm/causelybot/templates/bot_deployment.yaml:6]: this is redundant in helm

File: helm/causelybot/templates/bot_secret.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+kind: Secret
+metadata:
+  name: slack-webhook-secret
+  namespace:  {{ .Release.Namespace }}
+type: Opaque
+data:
+  webhook-url: {{ .Values.webhook.url | b64enc }}
+
+---
+apiVersion: v1
```
Comment (esara) [inline helm/causelybot/templates/bot_secret.yaml:13]: why do you need two secrets - these are used together

File: helm/causelybot/templates/bot_deployment.yaml (hunk 1 above)
Comment (sumanyumuku98) [reply helm/causelybot/templates/bot_deployment.yaml]: Addressed in the new commit!

File: helm/causelybot/templates/bot_secret.yaml (hunk 2 above)
Comment (sumanyumuku98) [reply helm/causelybot/templates/bot_secret.yaml]: Addressed in the new commit!

**Pull request merged**